data-processing/scripts/convert_dagw_to_jsonlgz.py

```diff
@@ -112,38 +114,65 @@ def reformat_dataset(ds: Dataset, num_proc: int) -> Dataset:
         "twfv19": "Twitter Folketingsvalget 2019 (Danish election tweets)",  # not present in this version of the dataset
     }
 
-    # update source
-    ds = ds.map(lambda x: {"source": source2longname[x["source"]]}, num_proc=num_proc)  # type: ignore
+    # update sub-source
+    ds = ds.map(lambda x: {"sub-source": source2longname[x["sub-source"]]}, num_proc=num_proc)  # type: ignore
 
-    # move license, domain to metadata
+    # move license, domain, sub-source to metadata
     ds = ds.map(  # type: ignore
-        lambda x: {"metadata": {"license": x["LICENSE"], "domain": x["domain"]}},  # type: ignore
+        lambda x: {"metadata": {"license": x["LICENSE"], "domain": x["domain"], "sub-source": x["sub-source"]}},  # type: ignore
         num_proc=num_proc,
     )
-    ds = ds.remove_columns(["LICENSE", "domain", "uri"])
-    return ds
+    ds = ds.remove_columns(["LICENSE", "domain", "uri", "sub-source"])
 
+    # add a fixed source: dagw
+    ds = ds.map(lambda x: {**x, "source": "dagw"})
 
-def main():
-    num_proc = 2
-    ds: DatasetDict = load_dataset("DDSC/partial-danish-gigaword-no-twitter")  # type: ignore
-    ds: Dataset = ds["train"]  # type: ignore
+    # filter out documents with empty "text" entries:
+    # either empty or totally while space
+    # 162051 out of 673211 empty docs found in total if skipping this step
+    ds = ds.filter(lambda x: x['text'] != '' and not x['text'].isspace(), num_proc=num_proc)
 
-    # reformat
-    ds = reformat_dataset(ds, num_proc=num_proc)
+    return ds
 
-    # save to jsonl.gz
-    ds.to_json("data.jsonl.gz", orient="records", lines=True, compression="gzip")  # type: ignore
+def filter_by_domains(ds: Dataset, domain_list: list) -> Dataset:
+    """
+    Filters the dataset to exclude records where the 'metadata' field's 'domain'
+    matches one of the domains specified in 'domain_list'.
+    """
+    # Filter dataset based on the domain list
+    return ds.filter(lambda x: x['metadata']['domain'] not in domain_list)
 
+def main():
+    num_proc = 32
+    # Domains to filter
+    domain_list = ['Danish daily newspapers', 'Common Crawl', 'Open Subtitles']
+
+    print("Start loading the dataset...")
+    ds: DatasetDict = load_dataset("DDSC/partial-danish-gigaword-no-twitter")
+    ds: Dataset = ds["train"]
+    print("Dataset loaded.")
+    print("Start reformatting the dataset...")
+    # Reformat and clean the dataset
+    ds = reformat_and_clean_dataset(ds, num_proc=num_proc)
+    print("Dataset reformatted and cleaned.")
+    # Filter dataset based on domains
+    ds = filter_by_domains(ds, domain_list)
+
+    # Print out a sample
+    sample = ds[0]
+    print("Sample Record:", sample)
+
+    print("Start saving the dataset...")
+    # Save to jsonl.gz
+    ds.to_json("data.jsonl.gz", orient="records", lines=True, compression="gzip")
```

Comment:
I believe this is only supported in the latest version of datasets so please update the dependency as well